pyro/distributions/multivariate_normal.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+import torch
+from torch.autograd import Variable
+
+from pyro.distributions.distribution import Distribution
+
+
+def potri_compat(var):
+    return Variable(torch.potri(var.data)) if torch.__version__ < '0.3.0' else torch.potri(var)
+
+
+class MultivariateNormal(Distribution):
+    """
+    Multivariate normal (Gaussian) distribution.
+
+    A distribution over vectors in which all the elements have a joint
+    Gaussian density.
+
+    :param torch.autograd.Variable mu: Mean. Must be a vector (Variable containing a 1d Tensor).
+    :param torch.autograd.Variable sigma: Covariance matrix. Must be symmetric and positive semidefinite.
+    :param is_cholesky: Should be set to True if you want to directly pass a Cholesky decomposition
+        of the covariance matrix as `sigma`.
+    :param use_inverse_for_batch_log: If this is set to true, the torch.inverse function will be used to compute
+        log_pdf. This means that the results of log_pdf can be differentiated with respect to sigma.
+        Since the gradient of torch.potri is currently not implemented, differentiation of log_pdf wrt. sigma is not
+        possible when using the Cholesky decomposition. Using the Cholesky decomposition is however much faster and
+        therefore enabled by default.
+    :raises: ValueError if the shape of mean or sigma is not supported.
+    """
+
+    reparameterized = True
+
+    def __init__(self, mu, sigma, batch_size=None, is_cholesky=False, use_inverse_for_batch_log=False, *args, **kwargs):
```

Comment:
Two comments on the interface:
1. It would be clearer to provide two alternative arguments `sigma=None` and `chol_sigma=None` and derive one from the other, rather than provide a single `sigma` and a flag `is_cholesky=False`. This is also more consistent with other distributions classes like `Bernoulli` and `Categorical` which can take either `probs=False` or `logits=False` and derives one from the other internally. I suppose the more common should be first, but I don't know which is more common.
2. Going forward we're trying to maximize compatibility with Tensorflow Distributions, so it would be preferable to rename `mu` -> `loc`, `sigma` -> `covariance_matrix`, and `sigma_cholesky` -> `scale_tril`. For reference, see https://www.tensorflow.org/versions/r1.0/api_docs/python/tf/contrib/distributions/MultivariateNormalCholesky and https://www.tensorflow.org/versions/r1.0/api_docs/python/tf/contrib/distributions/MultivariateNormalFull